### PR TITLE
Fix single-arg unnormalized parsing

### LINKED ISSUE #720: Error in *_lupdf syntax when there is just one argument
The 2.25 release candidate 1
These work
```
  z ~ std_normal();
  target += std_normal_lpdf(z);
  target += std_normal_lupdf(z|);
```
This should work, too
```
  target += std_normal_lupdf(z);
```
but now gives an error
```
Semantic error in '/tmp/Rtmp3VVrhb/model-61026b91fcfe.stan', line 36, column 12 to column 31:
   -------------------------------------------------
    34:  model {
    35:    // half-t priors for lambdas and tau, and inverse-gamma for c^2
    36:    target += std_normal_lupdf(z);
                     ^
    37:    target += student_t_lupdf(lambda | nu_local, 0, 1);
    38:    target += student_t_lupdf(tau | nu_global, 0, scale_global*2);
   -------------------------------------------------

Probabilty functions with suffixes _lpdf, _lupdf, _lpmf, _lupmf, _lcdf and _lccdf, require a vertical bar (|) between the first two arguments.
```

### PR DESCRIPTION
**Pull request by rok-cesnovar**
Fixes #720 

No release notes as this PR is fixing a bug for a feature that has not been released yet. 

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: src/frontend/parser.mly [hunk 1]
```diff
@@ -426,7 +426,9 @@ common_expression:
        if
          List.length args = 1
          && ( String.is_suffix ~suffix:"_lpdf" id.name
-            || String.is_suffix ~suffix:"_lpmf" id.name )
+              || String.is_suffix ~suffix:"_lupdf" id.name
+              || String.is_suffix ~suffix:"_lpmf" id.name
+              || String.is_suffix ~suffix:"_lupf" id.name )
```
Comment (avehtari) [inline src/frontend/parser.mly:431]: This should be `_lupmf`

Comment (avehtari) [reply src/frontend/parser.mly:431]: Hmmm.. Iäm not certain whether my comment is showing for the correct line, but `_lupf` should be `_lupmf`

Comment (rok-cesnovar) [reply src/frontend/parser.mly:431]: Its showing correctly. I was sloppy. Thanks.

**Review by avehtari — APPROVED** [review]
Looks good now. Great that you added tests, too.

**Pull request merged**